plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -0,0 +1,304 @@
+      githubCredentialsProvider?: GithubCredentialsProvider;
+    },
+  ) {
+    this.githubCredentialsProvider =
+      options.githubCredentialsProvider ||
+      SingleInstanceGithubCredentialsProvider.create(this.options.gitHubConfig);
+    this.scheduleFn = this.createScheduleFn(options.schedule);
+  }
+
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.getProviderName} */
+  getProviderName(): string {
+    return `github-entity-provider:${this.options.id}`;
+  }
+
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.connect} */
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    return await this.scheduleFn();
+  }
+
+  private createScheduleFn(schedule: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return schedule.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.options.logger.child({
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(error);
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: Logger) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const { org, repoSearchPath } = parseUrl(this.options.target);
+
+    const { headers } = await this.githubCredentialsProvider.getCredentials({
+      url: this.options.target,
+    });
+
+    const client = graphql.defaults({
+      baseUrl: this.options.gitHubConfig.apiBaseUrl,
+      headers,
+    });
+
+    const startTimestamp = Date.now();
+
+    logger.info(`Reading GitHub repositories from ${this.options.target}`);
+
+    const expectedEntityFiles = this.options.files;
+
+    const { repositories } = await getOrganizationEntities(
+      client,
+      org,
+      expectedEntityFiles,
+    );
+
+    const matching = repositories.filter(
+      r =>
+        !r.isArchived &&
+        repoSearchPath.test(r.name) &&
+        r.defaultBranchRef?.name,
+    );
+
+    const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+
+    logger.info(
+      `Read ${repositories.length} GitHub repositories (${matching.length} matching the pattern) in ${duration} seconds`,
+    );
+
+    const files = expectedEntityFiles.map(fileName => ({
+      name: fileName,
+      key: camelCase(fileName),
+    }));
+
+    const repoEntities: Array<LocationSpec | Array<LocationSpec>> = [];
+
+    for (const entity of matching) {
+      for (const { name: fileName, key } of files) {
+        const entityObj: any = entity[key as keyof typeof entity];
+        if (entityObj) {
+          const locationEntity = this.createLocationSpec({
+            url: entity.url,
+            branch: entity.defaultBranchRef?.name || '',
+            file: fileName,
+          });
```

Comment:
We need to support glob patterns still imo, just like the existing discovery processor. That means that if the filepath has a glob in it we can not use the `files` check and instead need to pass on the location as is for that repo, like the [processor does](https://github.com/backstage/backstage/blob/699b0cf3bd43ab8847a6fd7f44fb6705fcf0cfbf/plugins/catalog-backend-module-github/src/GithubDiscoveryProcessor.ts#L147-L156)